src/main/java/com/fasterxml/jackson/core/json/async/NonBlockingUtf8JsonParserBase.java

```diff
@@ -2169,12 +2170,13 @@ private final JsonToken _parseEscapedName(int qlen, int currQuad, int currQuadBy
 
         if (currQuadBytes > 0) {
             if (qlen >= quads.length) {
-                _quadBuffer = quads = growArrayBy(quads, quads.length);
+                _quadBuffer = quads = growArrayWithNameLenCheck(quads, quads.length);
             }
             quads[qlen++] = _padLastQuad(currQuad, currQuadBytes);
         } else if (qlen == 0) { // rare, but may happen
             return _fieldComplete("");
         }
+        _streamReadConstraints.validateNameLength(qlen << 2);
```

Comment:
Should be moved later, only check if `_symbols.findName()` can't find it.

Reply:
moved to _addName